test/actor.js

```diff
@@ -1070,6 +1070,119 @@ describe('Apify.pushRecord()', () => {
             });
     });
 });
+describe('Apify.getOrCreateStore()', () => {
+    it('throws on invalid args', () => {
+        const keyErrMsg = 'The "storeName" parameter must be a non-empty string';
+        expect(() => { Apify.getOrCreateStore(); }).to.throw(Error, keyErrMsg);
+        expect(() => { Apify.getOrCreateStore('', null); }).to.throw(Error, keyErrMsg);
+        expect(() => { Apify.getOrCreateStore('', 'some value'); }).to.throw(Error, keyErrMsg);
+        expect(() => { Apify.getOrCreateStore({}, 'some value'); }).to.throw(Error, keyErrMsg);
+        expect(() => { Apify.getOrCreateStore(123, 'some value'); }).to.throw(Error, keyErrMsg);
+    });
+
+    it('returns a Promise', () => {
+        const promise = Apify.getOrCreateStore('my-store');
+        expect(promise).to.be.a('promise');
+    });
+
+    const storeFabric = (nameOrId) => {
+        return new Promise((resolve, reject) => {
+            try {
+                resolve(Apify.getOrCreateStore(nameOrId));
+            } catch (error) {
+                reject(error);
+            }
+        });
+    };
+
+    const promiseStore = storeFabric('my-store');
+    it('returns an Object with a setValue and getValue method', () => {
+        promiseStore.then((store) => {
```

Comment:
I think this is invalid usage of promises in unit tests. Basically the function should return a new promise, the unit test then waits for it to resolve or fail. See other tests how to do that.